src/sensor_fusion.cpp

```diff
@@ -0,0 +1,97 @@
+
+/** @file source_fusion.cpp
+ * @brief The main function for the Sensor Fusion Project.
+ */
+
+// All C++ Headers
+#include <iostream>
+#include "../inc/sensor_fusion_common.hpp"
+#include "../inc/sensor_fusion_implementation.hpp"
+#include "../inc/fusion_algorithm.hpp"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// All C Headers go here
+
+#ifdef __cplusplus
+}
+#endif
+
+void print_usage() {
+    printf("Usage: filename contribution_p tolerance stuck_interval fusion_interval\n");
+    printf("\n");
+    printf("  filename              : The name of the file with the input values. (required)\n");
+    printf("  contribution_p        : The p parameter used in contribution rate selection.\n");
+    printf("  tolerance             : The tolerance ratio used in fault tolerance elimination process.\n");
+    printf("  stuck_interval        : The minimum interval where if a sensor doesn't change value it's considered stuck. (default = 10)\n");
+    printf("  fusion_interval       : The maximum interval where sensor values are fused. (default = 2)\n");
+    printf("\n");
+    exit(EXIT_SUCCESS);
+}
+
+int main(int argc, char *argv[]) {
+    setbuf(stdout, NULL);
+    printf("Program Name: %s\n", argv[0]);
+    printf("Number of arguments: %d\n", argc);
+    //TODO: ask user to provide column data in file
+
+    int ret = 0;
+    char *filename;
+    float contribution_p = 0.85, tolerance = 0.7;
```

Comment:
if these are unchanged values we need to define them as const and pass them as default in parse file in hpp